src/external_secondaries/validate_secondary_interface_test.cc

```diff
@@ -0,0 +1,111 @@
+#include <regex>
+#include <gtest/gtest.h>
+#include <boost/program_options.hpp>
+#include <boost/shared_ptr.hpp>
+
+namespace bpo = boost::program_options;
+
+std::string firmware;
+std::string target;
+
+std::string exec(const std::string &cmd) {
+  std::array<char, 128> buffer;
+  std::string result;
+  std::shared_ptr<FILE> pipe(popen(cmd.c_str(), "r"), pclose);
+  if (!pipe) throw std::runtime_error("popen() failed!");
+  while (!feof(pipe.get())) {
+    if (fgets(buffer.data(), 128, pipe.get()) != nullptr) result += buffer.data();
+  }
+  return result;
+}
+
+bool isVersionGood(const std::string &output){
+  return (output == FLASHER_VERSION || output == std::string(FLASHER_VERSION)+ "\n");
+}
+
+TEST(ECUInterface, check_version) {
+  std::string cmd = target + " api-version";
+  EXPECT_TRUE(isVersionGood(exec(cmd)));
+}
+
+TEST(ECUInterface, check_loglevel) {
+  std::string cmd = target + " api-version --loglevel 4";
+  EXPECT_TRUE(isVersionGood(exec(cmd)));
+}
+
+bool isECUListValid(const std::string &output){
+  std::smatch ecu_match;
+  std::regex ecu_regex("\\w+(?: \\w+)?\\n?");
+
+  unsigned int matched_symbols = 0;
+  std::string::const_iterator search_start(output.cbegin());
+  while(std::regex_search(search_start, output.cend(), ecu_match, ecu_regex)) {
+    matched_symbols += ecu_match.length();
+    search_start += ecu_match.position() + ecu_match.length();
+  }
+  return matched_symbols == output.size();
+}
```

Comment:
Since the goal here is to check that the output of the user's tool matches what Aktualizr is expecting when it calls the tool would it be better to put this parsing code somewhere it can be shared with the code inside aktualizr that will parse the data 'for real'?

Reply:
Good point, but where to put it? In utils.cc? Maybe later when we will need to use the same functionality we can move it to the appropriate place?

Reply:
Putting it in utils.cc is probably fine for now. Anton and I have a big secondaries branch that is still in progress, and until we merge that it's going to be difficult to pinpoint the right place.